### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by kain88-de**
Fixes #

Changes made in this Pull Request:
 - fix several warnings
 - remove usage of instant selectors in tests

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by kain88-de** [comment]
picks up the fixes from #1661

File: package/doc/sphinx/source/documentation_pages/overview.rst [hunk 1]
```diff
@@ -28,8 +28,8 @@ of a protein and the radius of gyration of the backbone atoms are calculated::
     from MDAnalysis.tests.datafiles import PSF,DCD  # test trajectory
     import numpy.linalg
     u = MDAnalysis.Universe(PSF,DCD)  # always start with a Universe
-    nterm = u.s4AKE.N[0]  # can access structure via segid (s4AKE) and atom name
-    cterm = u.s4AKE.C[-1]  # ... takes the last atom named 'C'
+    nterm = u.select_atoms('segid 4AKE and name N')[0]  # can access structure via segid (s4AKE) and atom name
+    cterm = u.selext_atoms('segid 4AKE and name C')[-1]  # ... takes the last atom named 'C'
```
Comment (jbarnoud) [inline package/doc/sphinx/source/documentation_pages/overview.rst]: There is a typo in `select_atoms`.

File: testsuite/MDAnalysisTests/analysis/test_hbonds.py [hunk 2]
```diff
@@ -73,7 +73,12 @@ def h(self, universe):
         kw = self.kwargs.copy()
         # kw.update(kwargs)
         h = MDAnalysis.analysis.hbonds.HydrogenBondAnalysis(universe, **kw)
-        h.run(verbose=False)
+        # remove in 1.0
+        if kw['detect_hydrogens'] == 'heuristic':
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/analysis/test_hbonds.py]: Thanks for remembering!

File: testsuite/MDAnalysisTests/core/test_segment.py [hunk 3]
```diff
@@ -69,12 +69,13 @@ def test_generated_residueselection():
     """Test that a generated residue group always returns a ResidueGroup (Issue 47)
     unless there is a single residue (Issue 363 change)"""
     universe = mda.Universe(PSF, DCD)
-    # only a single Cys in AdK
-    cys = universe.s4AKE.CYS
-    assert isinstance(cys, mda.core.groups.Residue), \
-        "Single Cys77 is NOT returned as a single Residue (Issue 47)"
+    with pytest.warns(DeprecationWarning):
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/core/test_segment.py]: I think this behavior will be removed in 1.0. right? Add a comment that this test will change??

Comment (jbarnoud) [inline testsuite/MDAnalysisTests/core/test_segment.py]: This test would be easier to find later if it had the same comment as the others about removing it before 1.0.

Comment (kain88-de) [reply testsuite/MDAnalysisTests/core/test_segment.py]: done above function definition.

**Pull request merged**